enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,463 @@
+
+### Alternative 1: OperatorStatus CRD
+
+In this approach, OLM would introduce a new namespaced [CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) called the OperatorStatus CRD. The OperatorStatus object would have no spec. The OperatorStatus's [Status subresource](https://book-v1.book.kubebuilder.io/basics/status_subresource.html) would include a map of conditions based off the  [Kubernetes APIServiceCondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podcondition-v1-core). The Maps key's will be consistent with the "Name" of the condition being communicated and the values.
+
+The OperatorStatus CRD would be used by operator to:
+- Communicate conditions and states that should influence how OLM manages the operator, such as preventing OLM from upgrading the operator.
+- Communicate messages to cluster admins, such as slow off-cluster queries or poorly configured CRs. These communications will not influence OLM behavior.
+
+Similarly, OLM may use the OperatorStatus CRD to:
+- Communicate to cluster admins the state of an operator, such as why an operator is unable to be upgraded.
+- Communicate with the operator about upcoming actions.
+
+
+#### Pros
+
+- A single point of communication between OLM and Operators.
+- Feature is opt-in, operators can choose not to take advantage of this feature.
+- This approach can be expanded to support new communication channels easily.
+- Allows communication of conditions that influence OLM or provide helpful messages to Cluster Admins.
+
+#### Cons
+
+- Operators are required to know when they are managed by OLM and write to an OLM resource.
+- Expands the scope of APIs introduced by OLM.
+- Could confuse users since the OLM owned Operator CRD already conveys status.
+
+
+### Alternative 2: Operators write to the status of the CSV or Operators API
+
+This approach considered allowing Operators to write to the status of the [CSV](https://github.com/operator-framework/api/blob/6468209c61b24bed76ed4701eba1a8b7f81b2377/crds/operators.coreos.com_clusterserviceversions.yaml#L7) or the [Operators](https://github.com/operator-framework/api/blob/6468209c61b24bed76ed4701eba1a8b7f81b2377/crds/operators.coreos.com_operators.yaml#L7) API rather than introducing a new API managed by OLM and was quickly abandoned for the reasons discussed below in Pros/Cons.
+
+#### Pros
+
+- No new APIs.
+- A single source of truth for operator status.
+- Could easily expand to support other communication channels.
+
+#### Cons
+
+- Operators are required to know when they are managed by OLM and write to an OLM resource.
+- Requires Operator Authors to learn OLM specific readiness reporting conventions.
+- Operators granted the ability to update a CSV's status can change fields which influences OLM behavior. For example, the operator might accidently change the `Phase` of a CSV.
```

Comment:
another con is that in general we try to avoid having multiple writers to a single resource (in this case you'd have OLM and the operator itself both writing to the CSV)